lib/llhttp/parser.js

```diff
@@ -149,127 +134,89 @@ class HTTPParser {
-      inst.exports.free(this[kBufferPtr])
-    }
-    this[kBufferSize] = size
-    this[kBufferPtr] = inst.exports.malloc(size)
-  }
-
-  [kGetHeaders] () {
-    const rawHeaders = []
-    for (let c = 0; c < this[kHeadersFields].length; c++) {
-      rawHeaders.push(this[kHeadersFields][c], this[kHeadersValues][c])
-    }
-    this[kHeadersFields] = []
-    this[kHeadersValues] = []
-    this[kHeaderSize] = 0
-    return rawHeaders
-  }
-
-  [kTrackHeader] (length) {
-    this[kHeaderSize] += length
-    if (this[kHeaderSize] >= this[kMaxHeadersSize]) {
-      inst.exports.llhttp_set_error_reason(this[kPtr], 'HPE_HEADER_OVERFLOW:Header overflow')
-      return constants.ERROR.USER
-    }
-    return 0
-  }
-
   [kOnMessageBegin] () {
     return 0
   }
 
-  [kOnHeaders] (rawHeaders) {}
-
-  [kOnHeadersComplete] (versionMajor, versionMinor, rawHeaders, method,
-    url, statusCode, statusMessage, upgrade, shouldKeepAlive) {
+  [kOnHeadersComplete] (rawHeaders, statusCode, upgrade, shouldKeepAlive) {
     return 0
   }
 
   [kOnBody] (body) {
     return 0
   }
 
-  [kOnMessageComplete] () {
+  [kOnMessageComplete] (rawTrailers) {
     return 0
   }
 
-  [kOnExecute] (ret) {}
+  [kOnUpgrade] (head) {}
 
   close () {
-    inst.exports.llhttp_free(this[kPtr])
-    this[kPtr] = null
-    inst.exports.free(this[kBufferPtr])
-    this[kBufferPtr] = null
-    this[kCurrentBuffer] = Buffer.alloc(0)
+    if (this[kPtr]) {
+      inst.exports.llhttp_free(this[kPtr])
+      this[kPtr] = null
+    }
+
+    if (this[kBufferPtr]) {
+      inst.exports.free(this[kBufferPtr])
+      this[kBufferPtr] = null
+    }
   }
 
   execute (data) {
-    currentParser = this
-    this[kCurrentBuffer] = data
+    assert(this[kPtr])
+
     // Be sure the parser buffer can contain `data`
     if (data.length > this[kBufferSize]) {
-      this[kMallocBuffer](Math.ceil(data.length / 4096) * 4096)
+      if (this[kBufferPtr]) {
+        inst.exports.free(this[kBufferPtr])
+      }
+      this[kBufferSize] = Math.ceil(data.length / 4096) * 4096
+      this[kBufferPtr] = inst.exports.malloc(this[kBufferSize])
+      // Instantiate a Unit8 Buffer view of the wasm memory that starts from the parser buffer pointer.
+      this[kBufferView] = new Uint8Array(inst.exports.memory.buffer, this[kBufferPtr], this[kBufferSize])
     }
-    // Instantiate a Unit8 Buffer view of the wasm memory that starts from the parser buffer pointer
-    // and has the size of `data`
-    const u8 = new Uint8Array(inst.exports.memory.buffer, this[kBufferPtr], data.length)
-    // Fill the view with `data`
-    u8.set(data)
+
+    this[kBufferView].set(data)
+
     // Call `execute` on the wasm parser.
     // We pass the `llhttp_parser` pointer address, the pointer address of buffer view data,
     // and finally the length of bytes to parse.
     // The return value is an error code or `constants.ERROR.OK`.
     // See https://github.com/dnlup/llhttp/blob/undici_wasm/src/native/api.c#L106
```

Comment:
I forgot to update this comments, now we have the src embedded, so we can point directly there. I'll do it in a PR.